### ISSUE TITLE
File.Report.Html doesnt generate index.html (NPE in connection clipping)

### ISSUE THREAD

**Issue by StephanAebi** [archimatetool/archi#431]
### Version of Archi, Operating System

Archi Version 3.3.1 Build 201811301135
Windows CPWin10 September 2017

### Expected Behaviour

File.Report.Html should create a HTML Index file

### Actual Behaviour

No HTML Index file is created. The directories and content are created.

### Steps to Reproduce the Behaviour
1. Open Model
2. Click on Model name
2. File.Report.Htm

[log.txt](https://github.com/archimatetool/archi/files/2770249/log.txt)
[TGF-Architecture.zip](https://github.com/archimatetool/archi/files/2770252/TGF-Architecture.zip)

**Comment by Phillipus** [comment]
@jbsarrodie Hi JB, this seems to be related to a patch you made in AbstractDiagramConnectionFigure#paintFigure(Graphics graphics)

If you open the attached model and the View "50 Architectural Roadmap" you'll see that the Viewpoint is set to grey out some connections and this leads to a NPE when rendering the graphics:

```
java.lang.NullPointerException
	at org.eclipse.draw2d.SWTGraphics.checkSharedClipping(SWTGraphics.java:338)
	at org.eclipse.draw2d.SWTGraphics.translate(SWTGraphics.java:1349)
	at org.eclipse.draw2d.Shape.paintFigure(Shape.java:125)
	at com.archimatetool.editor.diagram.figures.connections.AbstractDiagramConnectionFigure.paintFigure(AbstractDiagramConnectionFigure.java:271)
```

@StephanAebi As a workaround, set the Viewpoint to "none"

**Comment by Phillipus** [comment]
If a Viewpoint greys out a connection then it sets it disabled. See AbstractArchimateConnectionFigure line 45. Setting the connection disabled is leading to a NPE.

**Comment by jbsarrodie** [comment]
@Phillipus I'm looking at it atm...

**Comment by Phillipus** [comment]
Here's a simplified test model. Set the Viewpoint to "Migration" to reproduce.

[connection.zip](https://github.com/archimatetool/archi/files/2770419/connection.zip)

**Comment by Phillipus** [comment]
I can narrow this down to:

`graphics.setClip(p);`

**Comment by Phillipus** [comment]
A workaround could be to change line 264 to:

`if(isEnabled() && !l.isEmpty()) {`

**Comment by jbsarrodie** [comment]
> A workaround could be to change line 264 to:
> `if(isEnabled() && !l.isEmpty()) {`

Yes, but... I'd like to understand why this is happening (Eclipse in debug mode right now...)

**Comment by StephanAebi** [comment]
I deleted the Migration Model and it worked again.
Thank you!

**Comment by Phillipus** [comment]
BTW- this is not connected to report generation, but rendering the connection in a View.

**Comment by jbsarrodie** [comment]
This is an Eclipse bug in `SWTGraphics.checkSharedClipping()`: `currentState.relativeClip` should be checked for `null` but it isn't.

Now tryin to find a good workaround

**Comment by Phillipus** [comment]
Yes, it sets `currentState.relativeClip` to null in `SWTGraphics.setClip(Path)` line 1095:

```
appliedState.relativeClip = currentState.relativeClip = null;
```

It is called by `Shape.paintFigure(Graphics)` at line 124:

```
if (!isEnabled()) {
   graphics.translate(1, 1);
```

**Comment by Phillipus** [comment]
I created a monkey patch that seems to work around the NPE. Not sure if there are any side effects to this.

See temp branch "clippingpatch"

**Comment by Phillipus** [comment]
Refactored the monkey patch in branch "clippingpatch"...

@jbsarrodie Any good?

**Comment by Phillipus** [comment]
Fixed in Archi 4.3.2. Please re-open if the issue re-occurs.
